crates/chia-consensus/src/spendbundle_conditions.rs

```diff
@@ -240,17 +241,19 @@ mod tests {
         while let Some((spend, rest)) = a.next(all_spends) {
             all_spends = rest;
             // process the spend
-            let [parent_id, puzzle, amount, solution, _spend_level_extra] =
-                extract_n::<5>(&a, spend, ErrorCode::InvalidCondition).expect("extract_n");
-
+            let destructure_list!(parent_id, puzzle, amount, solution) =
+                <match_list!(BytesImpl<32>, Program, u64, Program)>::from_clvm(&a, spend)
+                    .expect("parsing CLVM");
             spends.push(CoinSpend::new(
                 Coin::new(
-                    a.atom(parent_id).as_ref().try_into().expect("parent_id"),
-                    tree_hash(&a, puzzle).into(),
-                    a.number(amount).try_into().expect("amount"),
+                    parent_id.try_into().expect("parent_id"),
+                    clvm_utils::tree_hash_from_bytes(puzzle.as_ref())
+                        .expect("hash")
+                        .into(),
```

Comment:
it's not obvious that this is an improvement. turning a single line of code into 3.

Reply:
Sure this one line is more verbose, but it actually saves similar casts on lines 253, 255, and 256. This only looks worse because `fmt` is splitting it across multiple lines

Reply:
If you import tree_hash_from_bytes it'd probably go back to one line 😉

Reply:
UPDATE: I have, and it has - and this change is still a net positive given lines saved elsewhere